lib/forms/item_types.js

```diff
@@ -119,16 +119,21 @@ function processInputSchema(comp, schema) {
 }
 
 function setComponentOptions(comp, options, setModelFunc) {
+    function trySetModelFunc(comp, data) {
+        if (! comp.isDestroyed()) setModelFunc(comp, data);
+    };
+
     if (options) {
         if (typeof options.then == 'function') {
-            setModelFunc(comp, [{ value: 0, label: 'loading...' }]);
+            trySetModelFunc(comp, [{ value: 0, label: 'loading...' }]);
             options
                 .then(
-                    function(data) { setModelFunc(comp, data); },
-                    function() { setModelFunc(comp, [{ value: 0, label: 'loading error' }]); }
+                    function(data) { trySetModelFunc(comp, data); },
```

Comment:
Any reason you're not going for arrow functions here, as it's a nice place for the shorter syntax and you're not working with the context at all.

Reply:
The original implementation used `function` syntax.  I didn't want to add 'noise' to this PR.